### DISCUSSION TITLE
Bonsai doesn't open after upgrading GPU

### DISCUSSION THREAD

**Discussion by KangByungwoo** [bonsai-rx/bonsai#1674] [Q&A]
Hi,

I found that Bonsai doesn't open after upgrading GPU. When I click on an existing file or new project as shown in the below screenshot, no new window appears. It has been working fine before the GPU upgrade, and although I'm not 100% sure that this is because of upgrading GPU, that's the only thing that I can think of that changed. Everything else seems to be working fine after upgrading GPU. 

![image](https://github.com/bonsai-rx/bonsai/assets/26239966/f49cede4-5dc6-4301-bcb0-cb519f267aeb)

Could you help me troubleshoot this issue? I'm a little hesitant to reinstall bonsai because getting the right version to install was a lot of headache... If I have to reinstall, how can I check the version of the bonsai and is there a way to transfer currently installed packages to the newly installed bonsai? Interestingly, I can open Manage Packages from the main menu as shown in the below screenshot.

![image](https://github.com/bonsai-rx/bonsai/assets/26239966/e5673164-18cb-4e3e-aefe-c6b7dfda2e19)

I am on Windows 10.

**Comment by bruno-f-cruz** [comment]
You can use environments to keep track of your package versions (https://bonsai-rx.org/docs/articles/environments.html).

I suspect that by changing your GPU you changed you monitor arrangements and the default window position is outside the bounds of your current monitor configuration. Simply delete `Bonsai.exe.settings` and you should be fine.

In the future, re-installing bonsai should always be an option, make sure you use environments :)

**Reply by KangByungwoo** [reply]
Thank you for your prompt response. Your suggestion worked! Also, could you answer my new questions in the edited post: how can I check the version of the bonsai and is there a way to transfer currently installed packages to the newly installed bonsai?

**Reply by bruno-f-cruz** [reply]
Check the folder of your bonsai.exe. It will be in that same folder.
Regarding your second question, I believe I already answered it. Was the environment tutorial not clear in a specific point that you need help with?

**Reply by KangByungwoo** [reply]
As for the first question, I am in that folder, but still I don't see a way to check the version of the bonsai.
![image](https://github.com/bonsai-rx/bonsai/assets/26239966/8f992f63-2d63-4d0a-b35d-3de2b565070d)

As for the second question, I'm not sure if the currently installed bonsai is the portable version that your link refers to, and thus if the suggestion there applies in my case.

**Reply by bruno-f-cruz** [reply]
Regarding the environment, as explained in the tutorial, the bonsai.config will contain all the information necessary to regenerate the environment (including your bonsai version). 

Here's a copy of the text for reference:

'''

The key to creating and updating environments is the Bonsai.config file, which keeps a record of all currently installed dependencies for a specific Bonsai setup. You can find this file in the same location of the Bonsai executable (Bonsai.exe). Anytime you install or update a package, Bonsai will automatically modify the config file.

The contents of the Bonsai.config file are compared with the current state of the Packages folder when Bonsai starts. If there are any missing packages the Bonsai bootstrapper will download them automatically to recover the expected state of the installation folder.
'''

Let me know if anything is not clear.

**Reply by KangByungwoo** [reply]
I see. So in a new computer, if I download the latest portable version of bonsai and copy the config file from my old bonsai I set up into that folder, and run bonsai.exe, will it not only download all the packages and their dependencies recorded in the config file, but also the same version of the bonsai as my old bonsai (i.e. will bonsai.exe downgrade itself to match the version in the config file)?

**Reply by KangByungwoo** [reply]
I opened the config file and it looks like this. Which variable refers to the version of bonsai.exe?
![image](https://github.com/bonsai-rx/bonsai/assets/26239966/ac8dad60-1d39-4bf4-98ae-200e916b8437)

**Reply by bruno-f-cruz** [reply]
`Bonsai` version 2.7.2. But again, the bonsai version will not fully define your dependencies.

**Reply by KangByungwoo** [reply]
I see. Just to clarify, will bonsai.exe in the portable version downgrade itself to match the version in the config file?

**Reply by bruno-f-cruz** [reply]
Depends on the exact version. Some executables will be compatible others won't. If you know what bonsai version you want to use you can modify the script shown in the tutorial to target that version. Check the setup.ps1 file example.

**Reply by KangByungwoo** [reply]
![image](https://github.com/bonsai-rx/bonsai/assets/26239966/233b3a14-c3da-45d3-a4ff-42c2b01888e8)
I followed your method to install bonsai in my new computer using both the latest version (2.8.1)  and original version I used (2.7.2). But as shown in the above screenshot, in both cases PointGrey2 Package could not be installed. Can you help me solve this issue? Currently I do not remember how I installed it in the first place, but would it be possible to directly copy the package from the old computer to the new computer?

**Reply by bruno-f-cruz** [reply]
Hey
Can you attach the bonsai.config you are trying to use?

**Reply by KangByungwoo** [reply]
[Bonsai_config.zip](https://github.com/bonsai-rx/bonsai/files/14441827/Bonsai_config.zip)
Here it is.

**Reply by bruno-f-cruz** [reply]
The assembly seems to be installed on my PC, but the package doesn't find all the necessary dependencies ( I don't have the pointgrey software installed so I cant test further). What version of the pointgrey vieweer/sdk do you have installed?

**Reply by KangByungwoo** [reply]
I found that I first have to install the pointgrey viewer before installing bonsai the way described above to have PointGrey2 Package installed. Thanks again for your help!
